samcli/commands/deploy/command.py

```diff
@@ -127,6 +127,13 @@
     help="Indicates whether to use JSON as the format for "
     "the output AWS CloudFormation template. YAML is used by default.",
 )
+@click.option(
```

Comment:
This is a good start, but we should ALSO plumb this into the package command. It's possible for the package command to be run on its own and that's where S3 bucket resolution will happen.